signup/signupController.js

```diff
@@ -106,19 +110,41 @@ define('signupController', [], function() {
 				"siteId": $scope.sharedProperties.selectedSite
 			};
 
-			SignupService.storeAppointments(data).then((response) => {
-				if (typeof response !== 'undefined' && response && response.success){
-					document.body.scrollTop = document.documentElement.scrollTop = 0;
-					$scope.appointmentId = response.appointmentId;
-					$scope.successMessage = $sce.trustAsHtml(response.message);
+			const firstName = $scope.data.firstName || '';
+			const lastName = $scope.data.lastName || '';
+			SignupService.getExistingClientAppointments(firstName, lastName).then((response) => {
+				if (typeof response !== 'undefined' && response && response.success) {
+					// response.numberExistingAppointments === false when query is empty because none exist
+					if(response.numberExistingAppointments === false || response.numberExistingAppointments['numberExistingAppointments'] === '0') {
+						$scope.hasExistingAppointment = false;
+					} else {
+						$scope.hasExistingAppointment = true;
+					}
 
-					// Send the confirmation email
-					if ($scope.data.email != null && $scope.data.email.length > 0) {
-						$scope.emailConfirmation();
+					// Only allow the client to sign up for an appointment if they don't currently have any uncancelled appointments
+					if(!$scope.hasExistingAppointment) {
+						SignupService.storeAppointments(data).then((response) => {
+							if (typeof response !== 'undefined' && response && response.success){
+								document.body.scrollTop = document.documentElement.scrollTop = 0;
+								$scope.appointmentId = response.appointmentId;
+								$scope.successMessage = $sce.trustAsHtml(response.message);
+		
+								// Send the confirmation email
```

Comment:
Should we start requiring email--unsure if we still have clients w/o email addresses? The new "Cancel/Reschedule Appointment" page's instructions don't make much sense unless the person actually had an email

Reply:
This is a great point--I figured less tech-literate clients would be able to email the address, but they might not have an email! And the phone number is buried at the bottom of the page. I'll bring this up to Dr. Moody for her judgement.